project4/name_server/bootstrap.cpp

```diff
@@ -49,43 +52,59 @@ void Bootstrap::HandleRequest(
   }
 }
 
+void Bootstrap::ReceiveAndHandle() {
```

Comment:
AFAIK, this is identical to the version from `Nameserver`, so I think you can just delete this.

Reply:
It listens to a different message type, so `Bootstrap` has to override it here.

Reply:
Ah, I see. If you wanted to get really fancy, you could make a templated protected version of this method and then just have the public method call it with the appropriate type parameter, but I'm not going to request that change.